PLUGIN_MIGRATION.md

```diff
@@ -0,0 +1,60 @@
+# Plugin Migration
+
+*Note*: This document is evolving and is in draft state.
+
+A goal of the Extensions SDK is to ease developer migration of plugins to extensions.  Supporting this goal:
+ - Plugin interfaces and their extension points are being migrated to equivalent Extension interfaces
+ - Same-named methods and arguments are used whenever possible, with differences highlighted in this migration guide
+ - Wrapper classes are provided when possible to retain existing code logic
+
+Migration of the [Anomaly Detection Plugin](https://github.com/opensearch-project/anomaly-detection) to an Extension is in progress to both identify and refine migration challenges and provide examples to other plugin developers.
+
+## Implement Extension Interfaces and Extension Points
+
+Change the implementing class `FooPlugin` to `FooExtension` and implement the `Extension` interface.
+ - Implement other corresponding Extension interfaces, for example `ActionPlugin` interface would be `ActionExtension`.
+ - Change extension point implementation, if necessary, to conform to new types. For example, `RestHandler` classes become `ExtensionRestHandler` classes.  Consider extending base classes such as `BaseExtensionRestHandler` to provide additional convenience methods.
+
+## Use Wrapper Classes
+
+### Replace ClusterService with SDKClusterService
+
+ - `clusterService.getClusterSettings().addSettingsUpdateConsumer()` works as is, but has a method taking a map parameter can do multiple updates more efficiently.
+ - `clusterService.state()` has no changes.
+
+### Replace Client with SDKJavaClient or SDKRestClient
+
+The `SDKClient` provides two client options.
+
+The `OpenSearchClient` will be supported, and is actively developed along with other language clients and should be used whenever possible.  This client does have significant implementation differences compared to existing `Client` API.
+
+The `SDKRestClient` provides wrapper methods matching the `Client` API (but not implementing it), implemented internally with the `RestHighLevelClient`.  While this speeds migration efforts, it should be considered a temporary "bridge" with followup migration efforts to the `SDKJavaClient` planned.
```

Comment:
A bit confused here and please ignore my naivety. 

**OpenSearchClient**  is mentioned on line 29. In line 31, you mention SDKJavaClient. Are they 2 different clients? In Line 29, did you intend to refer SDKJavaClient?

Reply:
> Are they 2 different clients? In Line 29, did you intend to refer SDKJavaClient?

Nope, my mistake. 

There are two clients but there is no `SDKJavaClient`.  `OpenSearchClient` (the `opensearch-java` client) is the main one being used.  The method to initialize it is `SDKClient.initializeJavaClient()` and I confused myself.  It is currently a wrapper around a `RestClient` but I believe the clients team is changing that.

Reply:
This helps. Thanks